src/figuresClasses.ts

```diff
@@ -1,19 +1,101 @@
 export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea: () => number;
+}
 
+function calcArea(area: number): number {
+  return Math.floor(area * 100) / 100;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: 'triangle' = 'triangle';
 
-}
+  color: 'red' | 'green' | 'blue';
+
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  constructor(
+    color: 'red' | 'green' | 'blue',
+    a: number,
+    b: number,
+    c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Invalid triangle');
+    }
 
-export class Circle {
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('Invalid triangle proportions');
+    }
 
+    this.a = a;
+    this.b = b;
+    this.c = c;
+    this.color = color;
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return calcArea(area);
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  color: 'red' | 'green' | 'blue';
+
+  radius: number;
+
+  shape: 'circle' = 'circle';
+
+  constructor(color: 'red' | 'green' | 'blue', radius: number) {
```

Comment:
You are duplicating code, take the type for color from Figure

Reply:
Sure, I have corrected this issue!